### PR TITLE
[FLINK-9580] Potentially unclosed ByteBufInputStream in RestClient##readRawResponse

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request fixed potentially stream connection leak*

## Brief change log

  - *Add finally block to close specific input stream*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by yanghua** [comment]
cc @zentol @pnowojski

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java [hunk 1]
```diff
@@ -292,30 +292,49 @@ private void readRawResponse(FullHttpResponse msg) {
 			ByteBuf content = msg.content();
 
 			JsonNode rawResponse;
+			InputStream in = null;
 			try {
-				InputStream in = new ByteBufInputStream(content);
+				in = new ByteBufInputStream(content);
```
Comment (pnowojski) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java]: couldn't you put this `InputStream` inside try with resource?

```
try (InputStream in = new ByteBufInputStream(content)) {
...
}
```

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java [hunk 2]
```diff
@@ -292,30 +292,49 @@ private void readRawResponse(FullHttpResponse msg) {
...
 				rawResponse = objectMapper.readTree(in);
 				LOG.debug("Received response {}.", rawResponse);
 			} catch (JsonParseException je) {
 				LOG.error("Response was not valid JSON.", je);
 				// let's see if it was a plain-text message instead
 				content.readerIndex(0);
+				ByteBufInputStream inputStream = null;
 				try {
-					ByteBufInputStream in = new ByteBufInputStream(content);
+					inputStream = new ByteBufInputStream(content);
```
Comment (pnowojski) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java]: ditto?

**Comment by yanghua** [comment]
cc @pnowojski

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**